tools/buildtools/packmol/packmol.xml

```diff
@@ -0,0 +1,196 @@
+params["packmol_format"]["packmol_inputs"][pmfile_index]["datasetname"] = "${s.input.name}"
+#end for        
+
+params["outfile"] = "$outfile" 
+
+import os
+from jinja2 import Environment, FileSystemLoader
+
+currentpath = "$__tool_directory__"  # should work generally
+template_environment = Environment(loader=FileSystemLoader(currentpath),lstrip_blocks=True, trim_blocks=True)
+template = template_environment.get_template('template.j2')
+print(params)
+
+with open("$parameteroutfile",'w+') as f:
+    f.write(template.render(params))
+
+os.system("packmol < "+"$parameteroutfile")
+
+            ]]>
+        </configfile>
+    </configfiles>
+    <inputs>
+        <conditional name="packmol_format">
+            <param name="selected_format" type="select" label="Select a file format:">
+                <option value="pdb" selected="true">pdb</option>
+                <option value="xyz">xyz</option>
+                <option value="tinker">tinker</option>
+                <option value="moldy">moldy</option>
+            </param>
+            <when value="pdb">
+                <expand macro="multiple_input" name="packmol_inputs" format="pdb" label="pdb input with molecule type" help_text="Select a pdb file in .pdb format"/>
+            </when>
+            <when value="xyz">
+                <expand macro="multiple_input" name="packmol_inputs" format="xyz" label="xyz input with molecule type" help_text="Select a xyz file in .xyz format"/>
+            </when>
+            <when value="tinker">
+                <expand macro="multiple_input" name="packmol_inputs" format="tinker" label="tinker input with molecule type" help_text="Select a tinker file in .tinker format"/>
+            </when>
+            <when value="moldy">
+                <expand macro="multiple_input" name="packmol_inputs" format="moldy" label="moldy input with molecule type" help_text="Select a moldy file in .moldy format"/>
+            </when>
+        </conditional>
+    <section name="allparams" title="General Parameters" expanded="false">
+        <param name="tolerance" type="float" value="2.0" label="distance tolerance" />
+        <param name="nloop" type="integer" value="20" label="maximum number of optimisation loops" />
+        <param name="maxit" type="integer" value="20" label="maximum number of gencan iterations per loop" />
+        <param name="seed" type="integer" value="-1" label="random number generator seed, default of -1 used the computer time to set the seed" />
+    </section>
+
+
+    </inputs>
+    <outputs>
+        <data format="pdb" name="outfile" label="${tool.name}: structure output">
+        <change_format>
+            <when input="selected_format" value="pdb" format="pdb"/>
+            <when input="selected_format" value="xyz" format="xyz"/>
+            <when input="selected_format" value="tinker" format="tinker"/>
+            <when input="selected_format" value="moldy" format="moldy"/>
```

Comment:
are these filytypes alls in Galaxy?

Reply:
only pdb at the moment - so those extras don't work yet. I can comment out?

Reply:
commenting out and committing in a sec